### ISSUE TITLE
data option not work in tests/lib/rules

### ISSUE THREAD

**Issue by Aladdin-ADD** [eslint/eslint#8559]
**Tell us about your environment**

* **ESLint Version:** master

**What did you do? Please include the actual source code causing the issue.**

```js
// file: tests/lib/rules/*.js
{ code: "a\n==\nb", errors: [{ message: "Expected '{{operator}}' and instead saw '=='.", data: { operator: "===" }, type: "BinaryExpression", line: 2 }] }
```

**What did you expect to happen?**
the message should be equal to "Expected '===' and instead saw '=='." 

**What actually happened? Please include the actual, raw output from ESLint.**
"Expected '{{operator}}' and instead saw '=='."

**Comment by not-an-aardvark** [comment]
This is working as intended -- the `data` for a message is currently part of a rule's internal API, so the only message the `RuleTester` sees is the string containing the interpolated data.
